src/esp/gfx/DrawableGroup.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+#include "DrawableGroup.h"
+#include "Drawable.h"
+
+namespace esp {
+namespace gfx {
+
+class Drawable;
+DrawableGroup& DrawableGroup::add(Drawable& drawable) {
+  registerDrawable(drawable);
+  this->Magnum::SceneGraph::DrawableGroup3D::add(drawable);
+  return *this;
+}
+
+DrawableGroup& DrawableGroup::remove(Drawable& drawable) {
+  unregisterDrawable(drawable);
+  this->Magnum::SceneGraph::DrawableGroup3D::remove(drawable);
+  return *this;
+}
+
+DrawableGroup::~DrawableGroup() {
+  // has to unregister any currently existing drawables
+  for (int iDrawable = 0; iDrawable < this->size(); ++iDrawable) {
+    unregisterDrawable(static_cast<Drawable&>((*this)[iDrawable]));
+  }
```

Comment:
With `attachedToGroup_` gone as I suggested above, this loop is no longer needed -- it would only empty the unordered_map, which we don't really care about since we're destructing anyway

Reply:
We actually care. I met the CI failure yesterday, due to this. The DrawableGroup (NOT your magnum lib) is deconstructed earlier than our Drawable (not your drawable), and thus I cannot call the `drawables()` function in the destructor of the Drawable! The corrade_assert fails!

In the dtor of the Drawable, I MUST call the drawables() to get the drawableGroup this drawable is currently in. So it can remove itself from that idToDrawable_ lookup table. Because there is a chance, we just delete a drawable during the runtime.

Simply put: I must handle both cases: DrawableGroup is deconstructed first, and a within Drawable is deconstructed first.

Reply:
This should like a bug we need to fix in our `DrawableGroup` then.  Looking at our code:

https://github.com/facebookresearch/habitat-sim/blob/fcb1a595b6c151b59a5c537738cac354cc7e91f8/src/esp/gfx/Drawable.cpp#L19-L25

If we made that
```cpp
DrawableGroup* Drawable::drawables() {
  auto drawables = Magnum::SceneGraph::Drawable3D::drawables();
  if (! drawables)
    return nullptr;
  CORRADE_ASSERT(
      dynamic_cast<DrawableGroup*>(drawables),
      "Drawable must only be used with esp::gfx::DrawableGroup!", {});
  return static_cast<DrawableGroup*>(drawables);
}
```

Then this would correctly return a nullptr, alleviating this issue.

Reply:
This is a smart idea! Will do it in this way.

Reply:
Ok, then if this assert is in place, `attachedToGroup_` ultimately *is* redundant, right?

Reply:
@mosra: Yes, correct. It was removed already.